### DISCUSSION TITLE
error: invalid new-expression of abstract class type ‘facebook::velox::dwrf::proto::orc::PostScript’

### DISCUSSION THREAD

**Discussion by Throwaway8** [facebookincubator/velox#2504] [General]
/usr/include/c++/11/bits/unique_ptr.h: In instantiation of ‘typename std::_MakeUniq<_Tp>::__single_object std::make_unique(_Args&& ...) [with _Tp = facebook::velox::dwrf::proto::PostScript; _Args = {facebook::velox::dwrf::proto::PostScript}; typename std::_MakeUniq<_Tp>::__single_object = std::unique_ptr<facebook::velox::dwrf::proto::PostScript>]’:
/home/throwaway/system/velox/./velox/dwio/dwrf/common/FileMetadata.h:66:50:   required from here
/usr/include/c++/11/bits/unique_ptr.h:962:30: error: invalid new-expression of abstract class type ‘facebook::velox::dwrf::proto::PostScript’
  962 |     { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/11/bits/unique_ptr.h: In instantiation of ‘typename std::_MakeUniq<_Tp>::__single_object std::make_unique(_Args&& ...) [with _Tp = facebook::velox::dwrf::proto::orc::PostScript; _Args = {facebook::velox::dwrf::proto::orc::PostScript}; typename std::_MakeUniq<_Tp>::__single_object = std::unique_ptr<facebook::velox::dwrf::proto::orc::PostScript>]’:
/home/throwaway/system/velox/./velox/dwio/dwrf/common/FileMetadata.h:73:55:   required from here
/usr/include/c++/11/bits/unique_ptr.h:962:30: error: invalid new-expression of abstract class type ‘facebook::velox::dwrf::proto::orc::PostScript’
cc1plus: all warnings being treated as errors
ninja: build stopped: subcommand failed.
make[1]: *** [Makefile:81: build] Error 1
make[1]: Leaving directory '/home/throwaway/system/velox'
make: *** [Makefile:88: release] Error 2

makes no sense is the dwrf proto files not compiled or something? I have protoc installed

**Comment by Throwaway8** [comment]
when trying to build velox

**Comment by pedroerp** [comment]
@Yuhta @yingsu00 since you guys are involved in that part of the codebase, any ideas here?

@Throwaway8 which compiler and version are you using?

**Comment by pedroerp** [comment]
Looks like there's a duplicate Issue associated:

> https://github.com/facebookincubator/velox/issues/2505
